auditexport/utils.py

```diff
@@ -108,39 +100,48 @@ def saveNextPageCursor(nextPageCursor, lastRecordSaved):
     with open(CURSOR_PATH, 'w') as cursorFile:
         cursorConfig.write(cursorFile)
 
+def pre_export(fileFormat, records):
+    logger.debug('exporting records to ' + fileFormat + '.....')
+    preparedRecords = [flattenObjectProp(record) for record in records]
+    return preparedRecords
+
 
 def exportToJson(pathToFolder, records):
-    logger.debug('exporting records to json.....')
-    preparedRecords = list(map(lambda record: flattenObjectProp(record), records))
+    preparedRecords = pre_export('json', records)
 
-    fileName = str(datetime.datetime.utcnow().isoformat()) + ".json"
+    fileName = str(datetime.datetime.utcnow().isoformat()) + '.json'
     fn = os.path.join(pathToFolder, fileName)
-    with open(fn, "w") as f:
+    with open(fn, 'w') as f:
         json.dump(preparedRecords, f, sort_keys=True,
                   indent=4, separators=(',', ': '))
 
 
-def exportToCsv(pathToFolder, records):
-    logger.debug('exporting records to csv.....')
-    preparedRecords = list(map(lambda record: flattenObjectProp(record), records))
+def exportToCsv(pathToFolder, records, writeHeaders):
+    preparedRecords = pre_export('csv', records)
 
     for record in preparedRecords:
         auditType = record['type']
-        fileName = auditType + ".csv"
-        __writeCsvFile(auditType, pathToFolder, fileName, record)
+        fileName = auditType + '.csv'
+        filePath = os.path.join(pathToFolder, fileName)
+
+        # Write header only once for each auditType
+        writeHeader = (
+            writeHeaders[record['type']]) = (
+                record['type'] not in writeHeaders)
```

Comment:
It might just be because I don't know much Python at all but this conditional seems to be a little bit complex, so I'm not sure what is being checked here. Can it be broken apart a bit to make it easier to understand?

Reply:
changed to `if...elif` structure